app/cdap/components/RecipeList/ViewAllRecipes.tsx

```diff
@@ -14,32 +14,104 @@
  * the License.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 import T from 'i18n-react';
-import RecipeList from 'components/RecipeList';
-
-import Box from '@material-ui/core/Box';
-import { SortBy, SortOrder } from './types';
-import Breadcrumbs from '@material-ui/core/Breadcrumbs';
+import styled from 'styled-components';
 import { Link } from 'react-router-dom';
+import Breadcrumbs from '@material-ui/core/Breadcrumbs';
 import Typography from '@material-ui/core/Typography';
+import Box from '@material-ui/core/Box';
 import { getCurrentNamespace } from 'services/NamespaceStore';
+import RecipeList from 'components/RecipeList';
+import { SortBy, SortOrder } from 'components/RecipeList/types';
+import RecipeDetails from 'components/RecipeManagement/RecipeDetails';
+import { IRecipe, ActionType } from 'components/RecipeList/types';
+import DrawerWidget from 'components/common/DrawerWidget';
+import ActionsPopover, { IAction } from 'components/shared/ActionsPopover';
+
+const ActionsWrapper = styled(Box)`
+  display: flex;
+  align-items: center;
+`;
 
 const PREFIX = 'features.WranglerNewUI.Recipe';
+const actions: IAction[] = [
+  {
+    label: T.translate(`${PREFIX}.edit`),
+  },
+  {
+    label: T.translate(`${PREFIX}.download`),
+  },
+  {
+    label: T.translate(`${PREFIX}.delete`),
+  },
+];
 
 const redirectToObj = `/ns/${getCurrentNamespace()}/wrangle`;
 
-const ViewAllRecipies = () => {
+export default function ViewAllRecipies() {
+  const [actionType, setActionType] = useState('');
+  const [isPanelOpen, setIsPanelOpen] = useState(false);
+  const [showPopover, setShowPopover] = useState(false);
+  const [recipe, setRecipe] = useState<IRecipe>();
+  const isViewRecipeAction = actionType === ActionType.VIEW_RECIPE;
   const viewRecipeHandler = (selectedObject: any) => {
-    // To do : Integrate view recipe details panel
+    toggleOpen();
+    setRecipe(selectedObject);
+    setActionType(ActionType.VIEW_RECIPE);
+  };
+
+  const togglePopover = () => {
+    setShowPopover(!showPopover);
   };
 
   const handleEditRecipe = (selectedObject: any) => {
     // To do : Integrate Edit recipe details panel
   };
 
+  const toggleOpen = () => setIsPanelOpen(!isPanelOpen);
+
+  const getChildComponent = () => {
```

Comment:
plz cross check this function
if both if conditions fail, what are we returning? void??
plz update!!!

Reply:
updated code by returning null once we integrate edit recipe based on condition check will return edit recipe component or null